### PR TITLE
Make use of Maven Mojo Annotations

### PR DESCRIPTION
**Pull request by ffbit**
Switched to Maven annotations

### CODE REVIEW

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java [hunk 1]
```diff
@@ -16,47 +18,27 @@
...
+    @Component
     private RepositorySystem repositorySystem;
 
-    /**
-     * The current repository/network configuration of Maven.
-     *
-     * @parameter default-value="${repositorySystemSession}"
-     * @readonly
-     */
+    @Component
```
Comment (carlspring) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: I wonder -- do you need both annotations (@Component and @Parameter) here? Wouldn't @Parameter suffice, or am I missing something?

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java [hunk 2]
```diff
@@ -16,47 +18,27 @@
...
+    @Parameter(defaultValue = "${repositorySystemSession}")
     private RepositorySystemSession repositorySession;
 
-    /**
-     * The project's remote repositories to use for the resolution.
-     *
-     * @parameter default-value="${project.remoteProjectRepositories}"
-     * @readonly
-     */
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}")
```
Comment (carlspring) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:41]: I think you missed the readonly parameter for the annotation here.

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java (hunk 1 above)
Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: That's github's view missed something ;)
Could you, please, take a look at full the full version of [file](https://github.com/ffbit/solr-maven-plugin/blob/5db81fb873330c51b52e911469381a1f7d3f33a2/src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java#L38)?

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java (hunk 2 above)
Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:41]: I agree, nice catch!
:+1: 

Thanks a lot!

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java (hunk 1 above)
Comment (carlspring) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: I just double-checked -- you have both annotations. I believe @Parameter would be enough.

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java (hunk 2 above)
Comment (carlspring) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:41]: Cheers! :)

Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:41]: Fixed :)

File: src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java (hunk 1 above)
Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: Unfortunately, both annotations are necessary in this particular case.

Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: Thanks for the check!

Comment (carlspring) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: A-ha! Okay then. My bad.
No problem! It's not like I am familiar with Solr, but I'm playing around with Lucene and elasticsearch, so I'll be keeping an eye on the project, just for kicks and to learn new things. I hope you won't mind my Maven-related comments/suggestions.

Comment (ffbit) [inline src/main/java/com/ffbit/maven/solr/AbstractSolrMojo.java:37]: Never mind, any feedback is welcome.

**Pull request merged**